consensus/src/dag/dag_driver.rs

```diff
@@ -47,20 +47,75 @@ pub enum DagDriverError {
     MissingParents,
 }
 
+/// Interface to decide if we should move forward to a new round
+#[async_trait]
+pub trait RoundState: Send {
+    async fn update(
+        &mut self,
+        highest_round: Round,
+        strong_links_for_highest_round: Option<Vec<NodeCertificate>>,
+    );
+    fn enter_new_round(&mut self, new_round: Round);
+
+    fn current_round(&self) -> Round;
+}
+
+pub struct OptimisticResponsiveRound {
```

Comment:
This is actually `PrevalentResponsive` round right?

Reply:
I never understand what that word means lol, we're going to use a different one anyway, will update soon